File: js/webdev_test.js

function add_five(x) {
    return x+5;
}
function add_three(x) {
    return x+3;
}
function add_four(x) {
    return x+4;
}
function multiply_by_three(x) {
    return x*3;
}
function multiply_by_seven(x) {
    return x*7;
}
function divide_by_three(x) {
    return x/3;
}
function divide_by_seven(x) {
    return x/7;
}
function subtract_five(x) {
    return x-5;
}
function subtract_three(x) {
    return x-3;
}


function solver(){
	var x = arguments[0];
    var output = new Array();
	for (var i = 1; i < arguments.length; i++) {
    	output[i-1] = arguments[i](x);
    }
    
	document.write('[' + output.toString().replace(/,/g, ', ') + ']' + '\n');
}

solver(3, add_three, add_five);
solver(12, add_five, subtract_three, multiply_by_seven, divide_by_three);

File: c++/webdev_test.cpp
#include <iostream>
#include <cstdarg>
#include <vector>
#include "functions.cpp"

using namespace std;

void solver (int num, int x, ...);

int main (void) {
    solver(2, 1, &add_five, &multiply_by_three);
    solver(5, 6, &add_five, &subtract_five, &multiply_by_three, &multiply_by_seven, &subtract_three);
    cin.ignore();
return 0;
}

void solver (int num, int x, ...) {
     va_list arguments;
     vector<int> output;
     va_start (arguments, x);
     int (*current)(int);
     for (int i = 0; i < num; i++) {
         current = va_arg(arguments, int (*)(int));
         output.push_back (current(x));
     }     
     va_end(arguments);
     cout<<"[";
     for (int i = 0; i < num-1; i ++) {
         cout<<output[i]<<", ";    
     }
     cout<<output[num-1]<<"]\n";
}
